### ISSUE TITLE
Technical spec for ability for a Livepeer.com user to cut off the stream's in progress session

### ISSUE THREAD

**Issue by wohlner** [livepeer/studio#283]
Currently, when a livepeer.com deletes a stream object, if a livestream is in progress, the stream continues.

**Comment by darkdarkdragon** [comment]
Possible implementation:

1. When API receives delete stream request, it goes to Consul and checks in what datacenter it is live now.
2. If it is in different datacenter it routes API request to one that have this stream (if stream is live in `fra`, but `lon` gets request to delete stream - it just hits fra.livepeer.com with the same request).
3. If it is live in same datacenter - it finds (from Consul) on which Mist instance it is running and hits Mist's API that stops the stream. Only then it updates DB and returns success to user.

**Comment by ericxtang** [comment]
Good thought @darkdarkdragon.  

I wonder if it makes sense to disallow the `delete` action for active streams (this is something that both [IVS](https://docs.aws.amazon.com/ivs/latest/APIReference/API_DeleteChannel.html) and Mux do).  I like the Mux approach a little better, because the user doesn't have to explicitly call `stop`, and instead, can call `reset` to stop the current stream AND create a new stream key.

**Comment by wohlner** [comment]
Yeah, definitely want to think through stopping a live stream session (and maybe prevent future sessions from starting) vs deleting the `stream` object . 

I'm glad we are brainstorming implementations. 👏🏽  @darkdarkdragon 

Let's not implement a solution quite yet though!

**Comment by iameli** [comment]
Hmmmmm, yeah. @darkdarkdragon gets at the difficulty here which is that Consul routes and Mist servers are region-local, but incoming API requests (whether that means "delete" or "stop" or whatever) can come in to any datacenter.

> it goes to Consul and checks in what datacenter it is live now.

This feels a little "magic" to me — it needs to look at the consul routes and infer that `mdw` can be accessed at the API server at `mdw.livepeer.com`. What about using the `region` record in the stream session in Postgres instead?

> If it is live in same datacenter - it finds (from Consul) on which Mist instance it is running and hits Mist's API that stops the stream. Only then it updates DB and returns success to user.

Similarly here — we could add an API record to a stream session that records the (cluster-local) address of the Mist server.

Basically... I like the idea but it feels more robust if we can use Postgres instead of Consul.

**Comment by iameli** [comment]
> Yeah, definitely want to think through stopping a live stream session (and maybe prevent future sessions from starting) vs deleting the stream object .

@wohlner Sounds like a new field to me, perhaps `enabled: true/false`? A stream with `enabled: false` wouldn't allow new streams in, and setting `enabled: false` could kick off any streams in progress.

**Comment by wohlner** [comment]
Let's create a new boolean parameter on the "parent" `stream` object (objects without a `parentId`) called `enabled`.

- [ ] the default is `true`
- [ ] all existing `stream` objects without a `parentId` should be backfilled to have this set to `true`
- [ ] when the `object` is created, return `enabled: true` in the response
- [ ] when `enabled` is set to 'false`: 1) don't allow any new sessions to begin and 2) halt any streams in progress.
- the live streamer experience in OBS for example should be the similar to if they lost connection mid-stream.
- no Livepeer.com dashboard UI update as part of this ticket. That will come in a future ticket
